src/main/java/sg/dex/starfish/keeper/DirectPurchaseAdapter.java

```
@@ -0,0 +1,111 @@
+        return tokenManager.tokenApprove(spenderAddress, price);
+    }
+
+    public boolean checkIsPaid(String purchaser_address, String publisher_address, BigInteger amount, String reference)
+    {
+        String purchaser_padded = Numeric.toHexStringWithPrefixZeroPadded(Numeric.toBigInt(purchaser_address), 64);
+        String publisher_padded = Numeric.toHexStringWithPrefixZeroPadded(Numeric.toBigInt(publisher_address), 64);
+        String reference_padded = Numeric.toHexStringWithPrefixZeroPadded(Numeric.toBigInt(reference), 64);
+        EthFilter filter = new EthFilter(DefaultBlockParameter.valueOf(BigInteger.valueOf(1)), DefaultBlockParameterName.LATEST, directPurchase.getContractAddress());
+
+        filter.addSingleTopic(EventEncoder.encode(directPurchase.TOKENSENT_EVENT));
+        // addOtionalTopic does not work for some reason.
+        filter.addSingleTopic(purchaser_padded);
+        filter.addSingleTopic(publisher_padded);
+        filter.addSingleTopic(reference_padded);
+
+        Flowable<DirectPurchase.TokenSentEventResponse> floable = directPurchase.tokenSentEventFlowable(filter);
+        ArrayList<DirectPurchase.TokenSentEventResponse> outcome = new ArrayList<>();
+        floable.subscribe(log -> {
+            outcome.add(log);
+        });
+
+        for (DirectPurchase.TokenSentEventResponse obj:outcome) {
+            if (obj._amount.equals(amount))
+                return true;
+        }
+        return false;
+    }
+
+    private Properties getProperties() {
+        Properties prop = new Properties();
+        try (InputStream input = getClass().getClassLoader().getResourceAsStream("application_spree.properties")) {
+
+            if (input == null) {
+                throw new IOException("properties files is missing");
+            }
+
+            prop.load(input);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Why are we consuming the exception here?

Have we actually recovered? If not then should re-throw the exception. (logging is fine of course)

---

whole private Properties getProperties() everywhere in the project will be reviewed and probably remade somehow @AyushVerma2